### PR TITLE
overwrite images

### PR DESCRIPTION
**Pull request by wolfganghuse**
**What this PR does / why we need it**:
Add `force_deregister` which will replace an existing destination image with the same name. The existing image will only be deleted if there is exactly one image with that name in PrismCentral

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
#35 

**How Has This Been Tested?**:
`make dev`
Run packer build with no existing destination image, exactly one image and multiple images existing.
**Special notes for your reviewer**:

**Release note**:

```release-note
add `force_deregister` to enable overwrite of existing destination images
```

### CODE REVIEW

File: builder/nutanix/config.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ type Config struct {
 	shutdowncommand.ShutdownConfig `mapstructure:",squash"`
 	ClusterConfig                  `mapstructure:",squash"`
 	VmConfig                       `mapstructure:",squash"`
+	ForceDeregister      			bool   `mapstructure:"force_deregister" json:"force_deregister" required:"false"`
```
Comment (tuxtof) [inline builder/nutanix/config.go:37]: fix format

File: builder/nutanix/driver.go [hunk 2]
```diff
@@ -656,6 +656,46 @@ func (d *NutanixDriver) SaveVMDisk(diskUUID string, imageName string) (*nutanixI
...
+	if ForceDeregister {
+		log.Println("force_deregister is set, check if image already exists")
+		ImageList, err:=conn.V3.ListAllImage(fmt.Sprintf("name==%s",imageName))
+		if err != nil {
+			return nil, fmt.Errorf("error while ListAllImage, %s", err.Error())
+		}
+		if *ImageList.Metadata.TotalMatches==0 {
+			log.Println("Image with given Name not found, no need to deregister")
+		}
+		if *ImageList.Metadata.TotalMatches>1 {
```
Comment (tuxtof) [inline builder/nutanix/driver.go:669]: else if

File: builder/nutanix/driver.go [hunk 3]
```diff
@@ -656,6 +656,46 @@ func (d *NutanixDriver) SaveVMDisk(diskUUID string, imageName string) (*nutanixI
...
+			log.Println("More than one mage with given Name found, will not deregister")
+		}
+		if *ImageList.Metadata.TotalMatches==1 {
```
Comment (tuxtof) [inline builder/nutanix/driver.go:672]: else if

File: builder/nutanix/driver.go [hunk 4]
```diff
@@ -656,6 +656,46 @@ func (d *NutanixDriver) SaveVMDisk(diskUUID string, imageName string) (*nutanixI
...
+			log.Println("Exactly one mage with given Name found, will deregister")
```
Comment (tuxtof) [inline builder/nutanix/driver.go:673]: s/mage/image/

File: builder/nutanix/driver.go [hunk 5]
```diff
@@ -656,6 +656,46 @@ func (d *NutanixDriver) SaveVMDisk(diskUUID string, imageName string) (*nutanixI
...
+		log.Println("force_deregister is set, check if image already exists")
+		ImageList, err:=conn.V3.ListAllImage(fmt.Sprintf("name==%s",imageName))
+		if err != nil {
+			return nil, fmt.Errorf("error while ListAllImage, %s", err.Error())
+		}
+		if *ImageList.Metadata.TotalMatches==0 {
+			log.Println("Image with given Name not found, no need to deregister")
+		}
+		if *ImageList.Metadata.TotalMatches>1 {
+			log.Println("More than one mage with given Name found, will not deregister")
```
Comment (tuxtof) [inline builder/nutanix/driver.go:670]: s/mage/image/

File: builder/nutanix/driver.go (hunk 2 above)
Comment (wolfganghuse) [reply builder/nutanix/driver.go]: fixed

File: builder/nutanix/driver.go (hunk 3 above)
Comment (wolfganghuse) [reply builder/nutanix/driver.go]: fixed

File: builder/nutanix/driver.go (hunk 4 above)
Comment (wolfganghuse) [reply builder/nutanix/driver.go]: fixed

File: builder/nutanix/driver.go (hunk 5 above)
Comment (wolfganghuse) [reply builder/nutanix/driver.go]: fixed

File: builder/nutanix/driver.go [hunk 6]
```diff
@@ -656,6 +656,44 @@ func (d *NutanixDriver) SaveVMDisk(diskUUID string, imageName string) (*nutanixI
...
+		} else if *ImageList.Metadata.TotalMatches==1 {
+			log.Println("Exactly one image with given Name found, will deregister")
+			resp,err:= conn.V3.DeleteImage(*ImageList.Entities[0].Metadata.UUID)
+			if err != nil {
+				return nil, fmt.Errorf("error while DeleteImage, %s", err.Error())
+			}
+			taskUUID := resp.Status.ExecutionContext.TaskUUID.(string)
+			log.Printf("Wait until delete Image %s is finished, %s\n",*ImageList.Entities[0].Metadata.UUID,taskUUID)
+			// Wait for the Image to be deleted
+			for {
```
Comment (tuxtof) [inline builder/nutanix/driver.go:679]: can you implement a timeout on the loop, here we can fall in an endless situation

File: example/source.nutanix.pkr.hcl [hunk 7]
```diff
@@ -17,7 +17,8 @@ source "nutanix" "centos" {
     subnet_name       = var.nutanix_subnet
   }
   
-  image_name        ="centos-{{isotime `Jan-_2-15:04:05`}}"
+  image_name        ="centos-packer-image"
```
Comment (tuxtof) [inline example/source.nutanix.pkr.hcl:20]: s/="/= "/g

**Review by tuxtof — CHANGES_REQUESTED** [review]

File: example/source.nutanix.pkr.hcl (hunk 7 above)
Comment (wolfganghuse) [reply example/source.nutanix.pkr.hcl]: fixed

File: builder/nutanix/driver.go (hunk 6 above)
Comment (wolfganghuse) [reply builder/nutanix/driver.go]: fixed

**Review by tuxtof — DISMISSED** [review]

**Review by tuxtof — APPROVED** [review]

**Pull request merged**